hs_extract/tests/content_types/test_raster.py

```diff
@@ -0,0 +1,120 @@
+    assert result_metadata["isPartOf"][
+        0].endswith("dataset_metadata.json")
+    assert result_metadata[
+        "user_metadata"] == "this is raster user metadata"
+
+
+@pytest.mark.skip(reason="User metadata event update for content types is not currently implemented")
+def test_raster_usermetadata():
+    resource_id = str(uuid.uuid4())  # Generate a random hex resource ID
+    print(resource_id)
+    with open("tests/test_files/rasters/single/logan1.tif", "rb") as f:
+        s3_client.upload_fileobj(
+            f, "test-bucket", f"{resource_id}/data/contents/raster_aggregation/logan1.tif")
+
+    # Wait for metadata to be consistent
+    sleep(1)
+    result_metadata = read_s3_json(f"test-bucket/{resource_id}/.hsjsonld/raster_aggregation/logan1.tif.json")
+    assert "user_metadata" not in result_metadata
+
+    write_s3_json(f"test-bucket/{resource_id}/.hsmetadata/raster_aggregation/logan1.tif.user_metadata.json", {
+                  "user_metadata": "this is raster user metadata"})
+    sleep(1)
+    result_metadata = read_s3_json(f"test-bucket/{resource_id}/.hsjsonld/raster_aggregation/logan1.tif.json")
+    assert result_metadata["user_metadata"] == "this is raster user metadata"
+
+
+def test_raster_vrt():
+    resource_id = str(uuid.uuid4())  # Generate a random hex resource ID
+    print(resource_id)
+    write_s3_json(f"test-bucket/{resource_id}/.hsmetadata/raster_aggregation/logan.vrt.user_metadata.json", {
+                  "user_metadata": "this is raster user metadata"})
+    sleep(1)
+    with open("tests/test_files/rasters/logan.vrt", "rb") as f:
+        s3_client.upload_fileobj(
+            f, "test-bucket", f"{resource_id}/data/contents/raster_aggregation/logan.vrt")
+    with open("tests/test_files/rasters/logan1.tif", "rb") as f:
+        s3_client.upload_fileobj(
+            f, "test-bucket", f"{resource_id}/data/contents/raster_aggregation/logan1.tif")
+    with open("tests/test_files/rasters/logan2.tif", "rb") as f:
+        s3_client.upload_fileobj(
+            f, "test-bucket", f"{resource_id}/data/contents/raster_aggregation/logan2.tif")
+
+    # Wait for metadata to be consistent
+    sleep(2)
+    # read in the resulting resource metadata file
+    result_resource_metadata = read_s3_json(
+        f"test-bucket/{resource_id}/.hsjsonld/dataset_metadata.json")
+
+    assert len(result_resource_metadata["associatedMedia"]) == 3
+    assert result_resource_metadata["associatedMedia"][0]["name"] in ["logan.vrt", "logan1.tif", "logan2.tif"]
+    assert len(result_resource_metadata["hasPart"]) == 1
+    assert result_resource_metadata["hasPart"][0]["url"].endswith("raster_aggregation/logan.vrt.json")
+
+    result_metadata = read_s3_json(
+        f"test-bucket/{resource_id}/.hsjsonld/raster_aggregation/logan.vrt.json")
+    assert len(result_metadata["associatedMedia"]) == 3
+    assert result_metadata["associatedMedia"][0]["name"] in ["logan.vrt", "logan1.tif", "logan2.tif"]
```

Comment:
In my earlier comment on multi file raster creation, I was pointing to the issue where the metadata extraction used either logan1.tif or logan2.tif file but included both these files as part of the associatedMedia of the single raster content type.